From e223dcd78da4356a2aebb2d99412229978ee4c12 Mon Sep 17 00:00:00 2001
Message-Id: <e223dcd78da4356a2aebb2d99412229978ee4c12.1685522257.git.stefan@agner.ch>
In-Reply-To: <f0292972b38d512fce48384ebbb72b2da46f03bd.1685522257.git.stefan@agner.ch>
References: <f0292972b38d512fce48384ebbb72b2da46f03bd.1685522257.git.stefan@agner.ch>
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 23 Feb 2023 09:34:12 +0000
Subject: [PATCH] Revert "amba_pl011: Don't use DT aliases for numbering"

This reverts commit c2fa32a8dae4c4b2638c9f4de5e398e98460e693.

Consistent, logical UART numbering is desirable, so remove the
effective revert.

See: https://forums.raspberrypi.com/viewtopic.php?t=347868

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 drivers/tty/serial/amba-pl011.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/tty/serial/amba-pl011.c b/drivers/tty/serial/amba-pl011.c
index 6f92f581e18c..d301253e26b1 100644
--- a/drivers/tty/serial/amba-pl011.c
+++ b/drivers/tty/serial/amba-pl011.c
@@ -2671,7 +2671,6 @@ static struct uart_driver amba_reg = {
 	.cons			= AMBA_CONSOLE,
 };
 
-#if 0
 static int pl011_probe_dt_alias(int index, struct device *dev)
 {
 	struct device_node *np;
@@ -2703,7 +2702,6 @@ static int pl011_probe_dt_alias(int index, struct device *dev)
 
 	return ret;
 }
-#endif
 
 /* unregisters the driver also if no more ports are left */
 static void pl011_unregister_port(struct uart_amba_port *uap)
@@ -2755,12 +2753,7 @@ static int pl011_setup_port(struct device *dev, struct uart_amba_port *uap,
 	if (IS_ERR(base))
 		return PTR_ERR(base);
 
-	/* Don't use DT serial<n> aliases - it causes the device to
-	   be renumbered to ttyAMA1 if it is the second serial port in the
-	   system, even though the other one is ttyS0. The 8250 driver
-	   doesn't use this logic, so always remains ttyS0.
 	index = pl011_probe_dt_alias(index, dev);
-	*/
 
 	uap->port.dev = dev;
 	uap->port.mapbase = mmiobase->start;
-- 
2.40.1

